week2/starter/ColorPicker/ContentView.swift

```diff
@@ -33,54 +33,82 @@
 import SwiftUI
 
 struct ContentView: View {
+  private let initialColor = Color(
+    red: Constants.Color.redInit / Constants.maxValue,
+    green: Constants.Color.greenInit / Constants.maxValue,
+    blue: Constants.Color.greenInit / Constants.maxValue
+  )
+  @Environment(\.verticalSizeClass) var verticalSizeClass
   @State private var alertIsVisible: Bool = false
-  @State private var redColor: Double = 0.0
-  @State private var greenColor: Double = 0.0
-  @State private var blueColor: Double = 0.0
-  @State private var foregroundColor = Color(red: 0, green: 0, blue: 0)
-
+  @State private var redColor: Double = Constants.Color.redInit
+  @State private var greenColor: Double = Constants.Color.greenInit
+  @State private var blueColor: Double = Constants.Color.blueInit
+  @State private var foregroundColor: Color
+  
+  init() {
+    _foregroundColor = State(initialValue: self.initialColor)
```

Comment:
This is unique, I have not seen it before in iOS development. It almost looks like you are intending for this to be like the .onAppear() view modifier.